intera_interface/src/motion_interface/motion_trajectory.py

```diff
@@ -0,0 +1,183 @@
+            if label is None: use default
+        @param joint_names: name for each joint in the robot
+            if joint_names is None: use default names
+        @param trajectory_options: options for the trajectory
+            if trajectory_options is None: use default options
+        """
+
+        # Used for sending the trajectory to the motion controller
+        self._client = MotionControllerActionClient()
+        self._traj = Trajectory()
+
+        self.set_label(label)
+        self.set_joint_names(joint_names)
+        self.set_trajectory_options(trajectory_options)
+
+    def stop_trajectory(self):
+        self._client.stop_trajectory()
+
+    def send_trajectory(self):
+        if not self._traj.waypoints:
+            rospy.logerr("Trajectory is empty! Cannot send.")
+            return None
+        self._client.send_trajectory(self.to_msg())
+        result = self._client.wait_for_result()
+        return result
+
+    def set_data(self, traj_msg):
+        if isinstance(traj_msg, Trajectory):
+            self._traj = deepcopy(traj_msg)
+        else:
+            rospy.logerr('Cannot set trajectory data. Invalid type.')
+
+    def set_label(self, label=None):
+        if label is None:
+            label = 'default'
+        self._traj.label = label
+
+    def set_joint_names(self, joint_names = None):
+        if joint_names is None:
+            joint_names = self.get_default_joint_names()
+        self._traj.joint_names = deepcopy(joint_names)
+
+    def set_trajectory_options(self, trajectory_options = None):
+        # TODO:  Better support for interaction control options.
+        if trajectory_options is None:
+            trajectory_options = TrajectoryOptions()
+            trajectory_options.interpolation_type = TrajectoryOptions.JOINT
+        if not isinstance(trajectory_options, TrajectoryOptions):
+            rospy.logerr('Cannot append options: invalid instance type!')
+        else:
+            self._traj.trajectory_options = deepcopy(trajectory_options)
+
+    def get_dimension(self):
+        """
+        @return: number of dimensions along the trajectory
+        """
+        return len(self._joint_names)
+
+    def get_label(self):
+        return self._traj.label
+
+    def clear_waypoints(self):
+        self._traj.waypoints = []
+
+    def append_waypoint(self, waypoint):
+        """
+        Appends a waypoint to the trajectory message
+        """
+        if isinstance(waypoint, Waypoint):
+            wpt = deepcopy(waypoint)
+        elif isinstance(waypoint, MotionWaypoint):
+            wpt = waypoint.to_msg()
+        else:
+            rospy.logerr('Invalid waypoint - cannot append to trajectory!')
+            return False
+
+        self._traj.waypoints.append(wpt)
+
+    def get_waypoint_joint_angles_as_list(self):
+        """
+        @return: a list of lists, with the following format:
+        [joint_names,waypoint_0,waypoint_1,...waypoint_N]
+        """
+        data = []
```

Comment:
I think [`extend`](https://docs.python.org/2/tutorial/datastructures.html) makes a lot of sense here
```
data = list()
data.append(self._traj.joint_names)
return data.extend([deepcopy(wpt.joint_positions) for wpt in self._traj.waypoints])
```